tardis/plugins/sqliteregistry.py

```diff
@@ -33,25 +34,48 @@ def __init__(self):
             for machine_type in getattr(configuration, site.name).MachineTypes:
                 self.add_machine_types(site.name, machine_type)
 
-    def add_machine_types(self, site_name: str, machine_type: str):
+    def add_machine_types(self, site_name: str, machine_type: str) -> None:
+        if self.get_machine_type(site_name, machine_type):
+            logger.debug(
+                f"{machine_type} is already present for {site_name} in database! Skipping insertion!"  # noqa B950
+            )
+            return
         sql_query = """
-        INSERT OR IGNORE INTO MachineTypes(machine_type, site_id)
+        INSERT OR ROLLBACK INTO MachineTypes(machine_type, site_id)
         SELECT :machine_type, Sites.site_id FROM Sites
         WHERE Sites.site_name = :site_name"""
         self.execute(sql_query, dict(site_name=site_name, machine_type=machine_type))
 
-    def add_site(self, site_name: str):
-        sql_query = "INSERT OR IGNORE INTO Sites(site_name) VALUES (:site_name)"
+    def get_machine_type(self, site_name: str, machine_type: str) -> List[Dict]:
+        sql_query = """
+        SELECT * FROM MachineTypes MT
+        JOIN Sites S ON MT.site_id = S.site_id
+        WHERE MT.machine_type = :machine_type AND S.site_name = :site_name"""
+        return self.execute(
+            sql_query, dict(site_name=site_name, machine_type=machine_type)
```

Comment:
Out of curiosity, why use a ``dict`` call instead of a ``{`` ``}`` literal? I don't mind either way since it's consistent with the rest of the class, though.

Reply:
Actually, there is no real reason to use `dict` instead of '{}'.

Reply:
Since it was not 100% consistent I changed it to `{}`.